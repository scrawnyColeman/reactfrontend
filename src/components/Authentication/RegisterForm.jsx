import React from 'react';
import TextField from '@material-ui/core/TextField';
import Button from '../Button/component';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
import { materialStyles } from '../../constants/styles';
import { useHistory } from 'react-router-dom';

const RegisterForm = () => {
    const history = useHistory();
    const useStyles = makeStyles(materialStyles);
    const classes = useStyles();

    return (
        <form className={classes.root} noValidate autoComplete="off">
            <Grid container spacing={3}>
                <Grid item xs={12}>
                    <TextField
                        id="outlined-username"
                        label="Username"
                        type="text"
                        variant="filled"
                        className={classes.textfield}
                    />
                </Grid>
            </Grid>
            <Grid container spacing={3}>
                <Grid item xs={12}>
                    <TextField
                        id="outlined-email"
                        label="Email Address"
                        type="email"
                        variant="filled"
                        autoComplete="off"
                        className={classes.textfield}
                    />
                </Grid>
            </Grid>
            <Grid container spacing={3}>
                <Grid item xs={12}>
                    <TextField id="outlined-dob" type="date" variant="filled" className={classes.textfield} />
                </Grid>
            </Grid>
            <Grid container spacing={3}>
                <Grid item xs={12}>
                    <TextField
                        id="outlined-password"
                        label="Password"
                        type="password"
                        variant="filled"
                        className={classes.textfield}
                    />
                </Grid>
            </Grid>
            <Grid container spacing={3}>
                <Grid item xs={12}>
                    <TextField
                        id="outlined-retyped-password"
                        label="Retype Password"
                        type="password"
                        variant="filled"
                        className={classes.textfield}
                    />
                </Grid>
            </Grid>
            <Grid container spacing={3}>
                <Grid item xs={12}>
                    <Button text="Register" variant="text" onClick={() => history.push({ pathname: 'home' })} />
                </Grid>
            </Grid>
        </form>
    );
};

export default RegisterForm;
