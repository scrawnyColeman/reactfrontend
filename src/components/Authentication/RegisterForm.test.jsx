import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import RegisterForm from './RegisterForm';

Enzyme.configure({ adapter: new Adapter() });

describe('RegisterForm component', () => {
    test('component matches snapshot', () => {
        const wrapper = shallow(<RegisterForm />);
        expect(wrapper).toMatchSnapshot();
    });
});
