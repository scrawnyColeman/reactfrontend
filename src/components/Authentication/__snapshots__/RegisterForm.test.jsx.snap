// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm component component matches snapshot 1`] = `
<form
  autoComplete="off"
  className="makeStyles-root-1"
  noValidate={true}
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(TextField))
        id="outlined-username"
        label="Username"
        onChange={[Function]}
        type="text"
        variant="filled"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(TextField))
        autoComplete="off"
        id="outlined-email"
        label="Email Address"
        onChange={[Function]}
        type="email"
        variant="filled"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(TextField))
        id="outlined-password"
        label="Password"
        onChange={[Function]}
        type="password"
        variant="filled"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(TextField))
        id="outlined-retyped-password"
        label="Retype Password"
        onChange={[Function]}
        type="password"
        variant="filled"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <Button
        onClick={[Function]}
        text="Register"
        variant="text"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</form>
`;
